util/src/keys/store.rs

```diff
@@ -256,6 +270,17 @@ impl SecretStore {
 		let unlock = try!(read_lock.get(account).ok_or(SigningError::AccountNotUnlocked));
 		Ok(unlock.secret as crypto::Secret)
 	}
+
+	/// Makes account unlocks expire and removes unused key files from memory
+	pub fn collect_garbage(&mut self) {
+		self.directory.collect_garbage();
+		let utc = UTC::now();
+		let expired_addresses = self.unlocks.read().unwrap().iter()
+			.filter(|&(_, unlock)| unlock.expires < utc)
+			.map(|(address, _)| address.clone()).collect::<Vec<Address>>();
+
+		for expired in expired_addresses { self.unlocks.write().unwrap().remove(&expired); }
```

Comment:
This does not remove anything from memory though. First there is no guarantee it is deallocated. Second, even if it is deallocated there is no guarantee memory will be cleared.